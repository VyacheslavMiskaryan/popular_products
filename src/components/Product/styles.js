import { makeStyles } from '@material-ui/core/styles';

import colorSet from '../../colors';

const styles = makeStyles(() => ({
  button: {
    height: 25,
    width: 25,
    color: colorSet.iconColor,
  },
}));

export default styles;
