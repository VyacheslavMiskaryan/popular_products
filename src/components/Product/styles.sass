@import '../../colors'
@import '../../global'

.grid-container
    display: grid
    grid-template: "img votes" 25px "img description" fit-content(100px) "img author" fit-content(50px) / 100px fit-content(700px)
    row-gap: 5px

.image-container
    grid-area: img

.votes-container
    grid-area: votes
    font-weight: bold

.description-container
    @extend %flex-container
    grid-area: description

    .title
        color: $title-color
        text-decoration: none

.author-signature-container
    display: flex
    align-items: center
    grid-area: author
    color: $signature-color