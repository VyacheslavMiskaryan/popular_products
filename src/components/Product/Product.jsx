import React, { useCallback } from 'react';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import { ButtonBase } from '@material-ui/core';
import ArrowDropUpIcon from '@material-ui/icons/ArrowDropUp';

import increaseProductRating from '../../redux/actions';
import styles from './styles';
import './styles.sass';

const Product = ({
  title,
  description,
  url,
  votes,
  submitterAvatarUrl,
  productImageUrl,
  itemIndex,
}) => {
  const dispatch = useDispatch();
  const classes = styles();

  const handleIncreaseRating = useCallback(() => {
    dispatch(increaseProductRating({ itemIndex }));
  }, [itemIndex]);

  return (
    <div className="grid-container">
      <div className="image-container">
        <img src={productImageUrl} alt="productAvatar" />
      </div>
      <div className="votes-container">
        <ButtonBase className={classes.button} onClick={handleIncreaseRating}>
          <ArrowDropUpIcon />
        </ButtonBase>
        {votes}
      </div>
      <div className="description-container">
        <Link to={`/${url}`} className="title">
          <span>{title}</span>
        </Link>
        <span>{description}</span>
      </div>
      <div className="author-signature-container">
        <span>Submitted by:</span>
        &ensp;
        <img src={submitterAvatarUrl} alt="avatar" />
      </div>
    </div>
  );
};

Product.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  votes: PropTypes.number.isRequired,
  submitterAvatarUrl: PropTypes.string.isRequired,
  productImageUrl: PropTypes.string.isRequired,
  itemIndex: PropTypes.number.isRequired,
};

export default Product;
