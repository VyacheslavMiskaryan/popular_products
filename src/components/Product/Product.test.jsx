import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import Product from './Product';
import store from '../../redux/store';

test('Products renders correctly', () => {
  const component = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Product
          title="Yellow Pail"
          description="On-demand sand castle construction expertise"
          url="/#"
          votes={44}
          submitterAvatarUrl="assets/images/avatarImages/daniel.png"
          productImageUrl="assets/images/productImages/image-aqua.png"
          itemIndex={1}
        />
      </BrowserRouter>
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
