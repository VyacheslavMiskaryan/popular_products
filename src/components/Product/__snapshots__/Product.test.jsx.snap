// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products renders correctly 1`] = `
<div
  className="grid-container"
>
  <div
    className="image-container"
  >
    <img
      alt="productAvatar"
      src="assets/images/productImages/image-aqua.png"
    />
  </div>
  <div
    className="votes-container"
  >
    <button
      className="MuiButtonBase-root makeStyles-button-1"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 14l5-5 5 5z"
        />
      </svg>
    </button>
    44
  </div>
  <div
    className="description-container"
  >
    <a
      className="title"
      href="//"
      onClick={[Function]}
    >
      <span>
        Yellow Pail
      </span>
    </a>
    <span>
      On-demand sand castle construction expertise
    </span>
  </div>
  <div
    className="author-signature-container"
  >
    <span>
      Submitted by:
    </span>
    â€‚
    <img
      alt="avatar"
      src="assets/images/avatarImages/daniel.png"
    />
  </div>
</div>
`;
